test/unit/carousel.ts

```diff
@@ -0,0 +1,403 @@
+import { utils } from '@storefront/core';
+import * as sinon from 'sinon';
+import Carousel from '../../src/carousel';
+import suite from './_suite';
+
+suite('Carousel', ({ expect, spy, stub }) => {
+  let carousel: Carousel;
+  let DEFAULT_SETTINGS;
+
+  describe('settings', () => {
+    it('should default to default settings if no settings are passed in', () => {
+      carousel = new Carousel();
+      const settings = {
+        speed: 0,
+        slidesToShow: 1,
+        slidesToScroll: 1
+      };
+
+      expect(carousel.state.settings).to.deep.eq(settings);
```

Comment:
Use `.eql()` instead of `.deep.eq()`. Only going to leave one comment to avoid spam, just find/replace deep.eq with eql.